src/hct_mis_api/apps/dashboard/services.py

```diff
@@ -201,3 +204,138 @@ def refresh_data(cls, business_area_slug: str) -> ReturnDict:
+    """
+    Utility class to manage global dashboard data caching using Redis.
+    """
+
+    @classmethod
+    def refresh_data(cls, business_area_slug: str = "global") -> ReturnDict:
+        """
+        Generate and store updated data for the global dashboard.
+        """
+        result = []
+        payments_aggregated = (
+            Payment.objects.using("read_only")
+            .select_related(
+                "business_area",
+                "program",
+                "delivery_type",
+                "financial_service_provider",
+                "parent",
+            )
+            .filter(
+                parent__status__in=["ACCEPTED", "FINISHED"],
+                is_removed=False,
+                conflicted=False,
+            )  # noqa
+            .exclude(status__in=["Transaction Erroneous", "Not Distributed", "Force failed", "Manually Cancelled"])
+            .annotate(
+                country=F("business_area__name"),
+                year=ExtractYear(Coalesce("delivery_date", "entitlement_date", "status_date")),
+                programs=Coalesce(F("household__program__name"), Value("Unknown program")),
+                sectors=Coalesce(F("household__program__sector"), Value("Unknown sector")),
+                fsp=Coalesce(F("financial_service_provider__name"), Value("Unknown fsp")),
+                delivery_types=F("delivery_type__name"),
+            )
+            .distinct()
+            .values(
+                "country",
+                "currency",
+                "status",
+                "year",
+                "programs",
+                "sectors",
+                "fsp",
+                "delivery_types",
+            )
+            .annotate(
+                total_usd=Sum(
+                    Coalesce("delivered_quantity_usd", "entitlement_quantity_usd", Value(0.0)),
+                    output_field=DecimalField(),
+                ),
+                total_payments=Count("id", distinct=True),
+                individuals=Sum(Coalesce("household__size", Value(1))),
+                households=Count("household", distinct=True),
+                children_counts=Sum("household__children_count"),
+                reconciled=Count("pk", distinct=False, filter=Q(payment_verifications__isnull=False)),
+                pwd_counts=pwdSum,
+                finished_payment_plans=finished_payment_plans,
+                total_payment_plans=total_payment_plans,
+            )
+        )
+
+        summary = defaultdict(
+            lambda: {
+                "total_usd": 0,
+                "total_payments": 0,
+                "sizes": 0,
+                "children_counts": 0,
+                "individuals": 0,
+                "households": 0,
+                "pwd_counts": 0,
+                "reconciled": 0,
+                "finished_payment_plans": 0,
+                "total_payment_plans": 0,
+            }
+        )
+
+        for item in list(payments_aggregated):
```

Comment:
iterator will get data in smaller chunks 
```suggestion
        for item in list(payments_aggregated.iterator()):
```